packages/snjs/lib/services/api/http_service.ts

```diff
@@ -26,10 +28,27 @@ export type HttpRequest = {
   authentication?: string;
 };
 
+type SNHttpServiceParams = {
+  environment: Environment;
+  appVersion: string
+};
+
 /**
  * A non-SNJS specific wrapper for XMLHttpRequests
  */
 export class SNHttpService extends PureService {
+  private readonly appVersion: string;
+  private readonly environment: Environment;
+
+  constructor({
+    environment,
```

Comment:
```suggestion
    private readonly environment: string,
```

This is shorthand for property notation in TS, and removes the need for declaring definitions on line 40 and setting values on line 49.

Reply:
I like this approach when arguments are provided one by one, but in our case constructor receives an object as an argument that contains `environment` and `appVersion`. Seems there isn't a good way to use shorthand notation for such cases.

Reply:
Object approach in service constructors would be the exception in our design. Probably no sense in modifying the pattern now. What's the benefit to objects in constructor if using TS? Usually they made sense in JS because you could make sure you're passing the right values regardless of order, but TS solves that problem.

Reply:
If the arguments count is small, then passing arguments one by one doesn't give problems. But if there are many (e.g., 4+) arguments, using objects does help. It's also very useful if some of those arguments are optional, since in non-object approach if you add new mandatory argument you need to either put it before the optional one or add empty value for optional argument.
But yeah, in this case we don't have any such case, so there is no any benefit from using object, so I updated it :)